qemu/tests/discard_for_raw_block_target.py

```diff
@@ -18,21 +20,27 @@ def run(test, params, env):
     2. Modprobe a 1G scsi_debug disk with writesame_16 mode.
     3. Trace the system calls while converting the zero image to the
        scsi_debug block device, then check whether 'fallocate' system call
-       works in the log.
+       results work in the log.
     :param test: Qemu test object.
     :param params: Dictionary with the test parameters.
     :param env: Dictionary with test environment.
     """
-    def check_output(strace_event, strace_output, match_str):
+
+    def _check_output(strace_event, strace_output, image_size, match_str):
         """Check whether the value is good in the output file."""
         logging.debug("Check the output file '%s'.", strace_output)
         with open(strace_output) as fd:
-            if match_str not in fd.read():
-                test.fail("The system call of '%s' is not right, "
+            m = re.findall(match_str + r', \d+, \d+', fd.read())
```

Comment:
Refer to previous code logic, do we need to add a check here, like
if not m:
test.fail("The system call of '%s' is not right, xxx)
Of course we'll hit an exception when m is empty(i.e. not find the match_str), but the error msg is not what we expect.
Next, if sum_size != byte_image_size, we can give another error msg, like xxx size is different from image size

If 'FALLOC_FL_PUNCH_HOLE in strace_output' is not an assertion in the test case, your code is OK for me; otherwise, it's better to have a check after the re.findall, IMO.

Reply:
Agree with you. And I have updated. Check again, please. Thanks